.travis.yml

```diff
@@ -1,8 +1,20 @@
 language: python
-python:
- - "2.7"
- - "3.6"
-install: pip install -r requirements.txt -r test-requirements.txt
-script:
-    - flake8 --filename="*.py"
-    - nosetests
+install:
+  - pip install -r requirements.txt -r test-requirements.txt
+cache:
+  directories:
+    - $HOME/.cache/pip
+matrix:
```

Comment:
Is there really no way to preserve the `python` section you removed and then remove the duplicate tests in the `matrix` section? Otherwise looks great :smile:

Reply:
Well, I could remove the `python` and `script` from the python2 tests and make it global, so any jobs in the matrix will inherit it by default, then only overwrite as needed but in my experience that its more confusing when looking at the travis file. It would look kind of like this:

```
language: python
python: "2.7"
script: nosetests
matrix:
  include:
    - name: "Python 2.7 tests"
    - name: "Python 2.7 flake8"
      script: flake8 --filename="*.py"
    - name: "Python 3.6 tests"
      python: "3.6"
    - name: "Python 3.6 flake8"
      python: "3.6"
      script: flake8 --filename="*.py"
```

Like, its really ugly and it brings confusion as where the commands come from, not really a fan.